src/hotspot/share/oops/method.cpp

```diff
@@ -1287,7 +1287,9 @@ void Method::link_method(const methodHandle& h_method, TRAPS) {
     h_method->_from_compiled_entry = SharedRuntime::get_handle_wrong_method_abstract_stub();
   } else if (_adapter == nullptr) {
     (void) make_adapters(h_method, CHECK);
+#ifndef ZERO
```

Comment:
With your change to sharedRuntime_zero.cpp I believe the preceding call to `make_adapters` will hit `ShouldNotReachHere()` and hence will not return. So, this #ifndef ZERO appears to be unnecessary.